### PR TITLE
Add Windows build

### PR DESCRIPTION
**Pull request by elasota**
Adds rudimentary native Windows compilation code path.
Fix configure script crash when running `python configure` in current directory.

### CODE REVIEW

**Comment by sfiera** [comment]
Should this also include the submodule update to `build/lib`?

**Comment by sfiera** [comment]
Also, configure:112 could be updated 🙂

**Comment by elasota** [comment]
Technically yes, but there are at least 3 other things necessary for it to actually work.  Would you rather hold off on this one til the rest of things are ready?

**Comment by sfiera** [comment]
There’s no harm in updating `gn-tools` already. I think it’s fine to start claiming Windows support, even if it’s incomplete. At worst, someone will try it, email me, and we’ll get a tester.

**Comment by sfiera** [comment]
Actually, it looks like I accidentally used non-master gn-tools, so I’ve got #390 to bring gn-tools back to master.

**Comment by sfiera** [comment]
With #390 merged, I restarted the tests here, but I expect it to be fine.

**Comment by elasota** [comment]
Updated this PR to just have all of the necessary updates... the biggest concern is if the script_executable change causes everything else to break.  Python installs on Windows are normally invoked as just `python` instead of `python3` but it looks like the `.gn` file has almost no context available to make that determination so hopefully it works in `BUILDCONFIG.gn`...

Otherwise this should work with the following build steps:
- Install Python 3
- Drop gn.exe and ninja.exe in the source directory
- Open a VS command prompt
- Change directory to the Antares source directory
- Run `python configure`
- Run `make-msvc.bat`

**Comment by elasota** [comment]
... and that failed.  Open to ideas.  I'm not very familiar with GN so the least-stupid idea I have at the moment is emitting `.gn` from `configure` if that variable really does need to be set there...

**Comment by sfiera** [comment]
I’m not sure. I see that [chromium uses python3.exe](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/.gn#13) and the [Windows build instructions](https://chromium.googlesource.com/chromium/src/+/HEAD/docs/windows_build_instructions.md) don’t seem to do anything special for it.

Emitting `.gn` from `./configure` seems fine for now.

**Comment by sfiera** [comment]
You can disable the cross-compiled build for now if you want.

**Comment by sfiera** [comment]
Is this ready for review now?

**Comment by elasota** [comment]
Yes, it should be set now.

File: src/glfw/video-driver.cpp [hunk 1]
```diff
@@ -514,7 +519,11 @@ void GLFWVideoDriver::loop(Card* initial) {
             main_loop.draw();
             glfwSwapBuffers(_window);
         } else {
+#ifdef _MSC_VER
             std::this_thread::sleep_for(std::chrono::microseconds(10));
```
Comment (elasota) [inline src/glfw/video-driver.cpp:523]: Apparently this is missing from mingw even though it's a C++11 standard function.

**Comment by sfiera** [comment]
Thanks!

**Pull request merged**